modules/native.nix

```diff
@@ -0,0 +1,34 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.typelevelShell.native;
+in
+{
+  options.typelevelShell.native = {
+    enable = mkEnableOption "Provide scala-native environment";
+
+    libraries = mkOption {
+      type = types.listOf types.package;
+      default = [ pkgs.zlib ];
```

Comment:
`zlib` is an optional dependency, but triggered by use of `java.util.zip`.  The only thing special about it is that it's highlighted by the Scala Native documentation.  By putting it here, it's easy to opt _out_ of, for those who want the leanest possible shell.

Reply:
> The only thing special about it is that it's highlighted by the Scala Native documentation.

I think it's the only optional lib needed by Scala Native itself. But not 100% on that.

Reply:
Seems like boehm, which could be a flag.  I don't know how to trigger it.  It does not appear to be autodetected.

Reply:
It's configured as an sbt setting `nativeGC := "boehm"`. Suffice to say, you won't stumble into it accidentally by calling some Javalib.

Reply:
Unless there's some environment variable that triggers that, I say we let people add the Nix library _and_ set it in their SBT.  I don't want to get too far down the rabbit hole of duplicating SBT settings in Nix.